packages/frontend/src/lib/__tests__/utils.test.ts

```
@@ -0,0 +1,78 @@
+import { generateFinalKey, generateOrderKey2, generateUserKey1 } from '../utils';
+
+describe('generateUserKey1', () => {
+  beforeAll(() => {
+    // Mock uniquement la méthode digest
+    if (!window.crypto) {
+      (window as any).crypto = {};
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)